multinet/api/views/workspace.py

```diff
@@ -173,3 +177,33 @@ def put_workspace_permissions(self, request, name: str):
             workspace.set_owner(new_owner)
 
         return Response(PermissionsReturnSerializer(workspace).data, status=status.HTTP_200_OK)
+
+    @swagger_auto_schema(query_serializer=AqlQuerySerializer())
+    @action(detail=True)
+    @require_workspace_permission(WorkspaceRoleChoice.READER)
+    def aql(self, request, name: str):
+        """Execute AQL in a workspace."""
+        query_str = request.query_params.get('query')
+        if query_str is None:
+            return Response('No query provided', status=status.HTTP_400_BAD_REQUEST)
+
```

Comment:
This is the real benefit of the serializer

```suggestion
		serializer = AqlQuerySerailizer(data=request.query_params)
		serializer.is_valid(raise_exception=True)
		query_str = serializer.validated_data['query']
```

As now, we don't need to manually throw a `400`, it will be thrown if `query` is not present.